simplesat/constraints/requirement.py

```diff
@@ -47,6 +48,41 @@ class Requirement(object):
     specs: seq
         Sequence of constraints
     """
+
+    @classmethod
+    def from_constraints(cls, constraint_tuple):
+        """ Return a Requirement object from a PackageMetadata constraint
+        tuple.
+
+        Parameters
+        ----------
+        constraints : constraints tuple
+            A tuple of constraints like (
+                'nose', ( # disjuncitons
```

Comment:
Determinism, not immutability.

It has to be done anyway for hashing to work and for things like pretty string printing to be deterministic. In JSON it can be an object instead of array, that's fine. In Python there's no particular reason to use a dict. We don't do arbitrary look-ups and we don't display it to anyone.